base/ca/src/main/java/com/netscape/cms/profile/input/ClientAuthInfoInput.java

```diff
@@ -0,0 +1,104 @@
+// --- BEGIN COPYRIGHT BLOCK ---
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; version 2 of the License.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+// (C) 2007 Red Hat, Inc.
+// All rights reserved.
+// --- END COPYRIGHT BLOCK ---
+package com.netscape.cms.profile.input;
+
+import java.util.Locale;
+import java.util.Map;
+
+import com.netscape.certsrv.profile.EProfileException;
+import com.netscape.certsrv.property.Descriptor;
+import com.netscape.certsrv.property.IDescriptor;
+import com.netscape.cms.profile.common.Profile;
+import com.netscape.cms.profile.common.ProfileInputConfig;
+import com.netscape.cmscore.apps.CMS;
+import com.netscape.cmscore.request.Request;
+
+/**
+ * This class implements the client authentication information
+ * input that collects RA client information such as uid, name,
```

Comment:
It's good that "RA" is mentioned in the comment here.  However, I'm wondering if we need to make it more clear by changing the name of this input plugin code as well as other parameters to make it more clear what these are for.  Would something like estClientAuthInfoInput.java (or "raClientAuthInfoInput.java") do? And perhaps any "external facing" ones such as the UID, NAME, and CERT defined below (e.g. est_client_uid, est_client_name, est_client_certificate)?  No need to change the internal names "UID" etc. Just the definition ("client_uid") that's being used externally.  Use "ra" instead of "est" if they are likely to be shared by other types of RAs.

Reply:
and CMS_PROFILE_CLIENT_UID -> CMS_PROFILE_EST_CLIENT_UID and the other two params, etc.

Reply:
need @edewata 's opinion too.

Reply:
Yes, I will modify the name to with RA prefix since this could be not limited to EST.